### PR TITLE
Add SimpleITK_PYTHON_USE_USE_LIMITED_API option

### PR DESCRIPTION
**Pull request by blowekamp**
More information on the Python Stable API:
https://docs.python.org/3/c-api/stable.html

Requires Python 3.11+, SWIG 4.2+, and CMake 3.26+.

### CODE REVIEW

File: Wrapping/Python/Python.i [hunk 1]
```diff
@@ -19,6 +19,10 @@
 
 %begin %{
 #include "sitkPython.h"
+// Needed by SWIG, but not in Stable Python API.
+// https://github.com/swig/swig/pull/2856
+#define NOMINMAX
```
Comment (blowekamp) [inline Wrapping/Python/Python.i:24]: @thewtex I had the weird error messages caused by MSVC headers defining min and max. I saw your PR for SWIG do you think this should go into SWIG too?

**Review by dave3d — APPROVED** [review]

File: Wrapping/Python/CMakeLists.txt [hunk 2]
```diff
@@ -4,7 +4,23 @@ cmake_minimum_required(VERSION ${SITK_CMAKE_MINIMUM_REQUIRED_VERSION})
 project( SimpleITK_Python )
 
 include(../../CMake/sitkProjectLanguageCommon.cmake NO_POLICY_SCOPE)
-find_package( Python REQUIRED COMPONENTS Interpreter Development.Module )
+
+find_package( Python REQUIRED COMPONENTS Interpreter Development.Module OPTIONAL_COMPONENTS Development.SABIModule)
+set( _PYTHON_DEVELOPMENT_MODULE "Development.Module" )
+if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.26" AND
+    SWIG_VERSION VERSION_GREATER_EQUAL "4.2.0" AND
+    Python_VERSION VERSION_GREATER_EQUAL "3.11.0") #
```
Comment (zivy) [inline Wrapping/Python/CMakeLists.txt:12]: Please remove the # at end of line.

File: Wrapping/Python/CMakeLists.txt [hunk 3]
```diff
@@ -19,6 +35,24 @@ mark_as_advanced( SimpleITK_PYTHON_EGG )
 option ( SimpleITK_PYTHON_WHEEL "Add building of python wheels to the dist target." ON )
 mark_as_advanced( SimpleITK_PYTHON_WHEEL )
 
+if(SWIG_VERSION VERSION_GREATER_EQUAL "4.2.0" AND
+        Python_VERSION VERSION_GREATER_EQUAL "3.11.0") # CMake atleast 3.26
```
Comment (zivy) [inline Wrapping/Python/CMakeLists.txt:39]: Possibly change comment to cmake_minimum_required > 3.26 (referencing the macro?) not sure about this.

File: Wrapping/Python/CMakeLists.txt [hunk 4]
```diff
@@ -19,6 +35,24 @@ mark_as_advanced( SimpleITK_PYTHON_EGG )
...
+
+option( SimpleITK_PYTHON_USE_LIMITED_API "Use Python limited API, for minor version compatibility." ${_SimpleITK_PYTHON_USE_LIMITED_API_DEFAULT} )
+mark_as_advanced( SimpleITK_PYTHON_USE_LIMITED_API )
+
+if (SimpleITK_PYTHON_USE_LIMITED_API AND NOT _SimpleITK_PYTHON_USE_LIMITED_API_DEFAULT)
+  message(WARNING "Python limited API is not supported for this Python version and SWIG version.")
+  message(WARNING "SWIG_VERSION: ${SWIG_VERSION}")
+  message(WARNING "Python_VERSION: ${Python_VERSION}")
+  message(WARNING "Development.SABIModule_FOUND: ${Development.SABIModule_FOUND}")
+  set(SimpleITK_PYTHON_USE_LIMITED_API OFF)
```
Comment (zivy) [inline Wrapping/Python/CMakeLists.txt:53]: Please add a message specifying the minimal version requirements for SWIG and Python. 

Not sure about this, should this be an error? User turned SimpleITK_PYTHON_USE_LIMITED_API on but package versions aren't valid.

**Review by zivy — APPROVED** [review]
Some minor comments.

File: Wrapping/Python/CMakeLists.txt (hunk 4 above)
Comment (blowekamp) [reply Wrapping/Python/CMakeLists.txt:44]: done

File: Wrapping/Python/Python.i (hunk 1 above)
Comment (thewtex) [reply Wrapping/Python/Python.i:24]: @blowekamp I recall something like this a long time ago, but I have not seen it since then.

**Pull request merged**